r4d4m4n71s/Learning.ReportPortal
# Test runners

Test Runners Task #2
https://epam-my.sharepoint.com/:w:/p/yury_karpinski/EeJgAVv9goFNgmTPK2i1t-YBSL9QjCJCbKESgu2Sdm5l2Q?e=t9mLK6

* Change xunit to mstest test runner since the last pull request
* Use csv data provider to implement BDD extending mstest ITestDataSource 
https://github.com/Microsoft/testfx-docs/blob/main/RFCs/007-DataSource-Attribute-VS-ITestDataSource.md
* Add Launch and Launces POM and business functions to compare expected data vs displayed grid.  TestLaunches_Ok and TestLaunchXSuites_Ok.
* For parallel execution was used .runsettings.  
Note: For number of threads updating there is 2 approaches:  updating .runsettings file or extending Assembly attribute to use a custom setup(cumbersome).. 

https://github.com/Microsoft/testfx-docs/blob/main/RFCs/004-In-Assembly-Parallel-Execution.md

taking in account the thread handling is test runner domain then the best choice is modifying the .runsettings file previous to the execution.  It could be achieve opening the file (.runsettings) and updating the number of threads field.
